### PR TITLE
Add operation log related interfaces

### PR DESCRIPTION
**Pull request by wu-clan**
There is a problem with this middleware that needs to be addressed.

WIP.

### CODE REVIEW

**Comment by wu-clan** [comment]
I have tried some efforts, but there seems to be a problem with using request bodies in middleware for Starlette version 0.27.0, see https://github.com/encode/starlette/pull/2026 for details.

**Comment by wu-clan** [comment]
I think it is possible to merge this PR ahead of time, provided that the problematic code is commented out and a new PR is opened to fix it after Starlette is updated.

**Comment by wu-clan** [comment]
Hi, @downdawn

Any ideas?

File: backend/app/middleware/opera_log_middleware.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        code: int = 200
+        msg: str = 'Success'
+        status: bool = True
+        err: Any = None
+
+        # 执行请求
+        start_time = datetime.now()
+        try:
+            await self.app(request.scope, request.receive, send)
+            log.info('3')
```
Comment (downdawn) [inline backend/app/middleware/opera_log_middleware.py:84]: The output here seems meaningless

**Comment by downdawn** [comment]
No problem.

It is recommended to keep MIDDLEWARE_ACCESS.

**Comment by downdawn** [comment]
Some native methods.

```
user_agent = request.headers.get('User-Agent')
_user_agent = parse(user_agent)
device = _user_agent.get_device()
os = _user_agent.get_os()
browser = _user_agent.get_browser()
```

**Review by downdawn — APPROVED** [review]

**Comment by wu-clan** [comment]
> 

I will restore it.

**Comment by wu-clan** [comment]
> No problem.
> 
> I suggest keeping the MIDDLEWARE_ACCESS because this middleware has become complex now and there may be some unknown issues.

I will restore it.

**Comment by wu-clan** [comment]
> Some native methods.
> 
> ```
> user_agent = request.headers.get('User-Agent')
> _user_agent = parse(user_agent)
> device = _user_agent.get_device()
> os = _user_agent.get_os()
> browser = _user_agent.get_browser()
> ```

No problem at all.

**Comment by wu-clan** [comment]
TBR.

**Pull request merged**